include/bit_algorithm_details.hpp

```diff
@@ -132,6 +132,119 @@ ForwardIt word_shift_right(ForwardIt first,
     std::fill(first, it, 0); 
     return it;
 }
+
+// returns a word consisting of all one bits 
+constexpr auto _all_ones() {
+    return -1;
+} 
+
+// returns a word consisting of all zero bits
+constexpr auto _all_zeros() {
+    return 0;
+}
+
+// checks that the passed iterator points to the first bit of a word
+template <class It>
+bool _is_aligned_lsb(bit_iterator<It> iter) {
+    return iter.position() == 0;
+}
+
+// checks that maybe_end is one position past the last bit of base
+template <class It>
+bool _is_one_past_last_bit(bit_iterator<It> maybe_end, It&& base) {
+    return maybe_end.position() == 0 
+      && std::next(std::forward<It>(base)) == maybe_end.base();
+}
+
+// checks that two bit iterators point to the same word
+template <class It>
+constexpr bool _in_same_word(bit_iterator<It> lhs, bit_iterator<It> rhs) {
+    return lhs.base() == rhs.base();
+}
+
+// simple alias for right shift
+template <class WordType>
+WordType _shift_towards_lsb(WordType word, std::size_t n) {
+    return word >> n; 
+} 
+
+// simple alias for left shift
```

Comment:
can we wrap the two shifts in static_cast?

Reply:
are you sure we need to? was curious so I tried it https://godbolt.org/z/fr8mZ6